jamaljsr/polar
# feat(tapd): Add support for litd v0.14.0-alpha.rc1

### Description

Adds support for the upcoming `litd` v0.14.0-alpha release. There were some changes necessary on the Polar side to get it working correctly. This mostly focused on the new `tapd` [AddInvoice](https://lightning.engineering/api-docs/api/taproot-assets/taproot-asset-channels/add-invoice) and [SendPayment](https://lightning.engineering/api-docs/api/taproot-assets/taproot-asset-channels/send-payment) RPCs which simplify the integration of creating and paying asset invoices. Polar previously used had to re-implement the code from `litcli ln addinvoice/payinvoice`. By switching to use these new RPCs, I was able to remove a bunch of lower level RPCs calls.

I have been testing this with the PR https://github.com/lightninglabs/lightning-terminal/pull/848 which hasn't been merged yet. So I will keep this PR open until the final `litd` v0.14.0 version is released.

### Steps to Test

First, you'll need to build the docker image using the litd PR branch:
```
$ cd docker/litd/
$ docker build --build-arg LITD_VERSION=update-to-lnd-18-4 -t polarlightning/litd:0.14.0-pre1 -f src.Dockerfile --no-cache .
```

Then in Polar:
1. Create a new network with 4 `litd` nodes
2. Mint a PUSD asset with a 1000 supply
3. Create an 500 PUSD asset address on `carol`
4. Send PUSD to that address from `alice`
5. Open a 400 PUSD asset channel from `alice` to `bob`
6. Open a 1M sats channel from `bob` to `carol`
7. Open a 400 PUSD asset channel from `carol` to `dave`
8. Create a 200 PUSD asset invoice on `dave`
9. Pay the invoice on `alice` using PUSD
10. Confirm the payment completes successfully

### Screenshots

![image](https://github.com/user-attachments/assets/1347ad09-b3e0-4b82-ac12-584a6a5f9714)
